packages/react-native-reanimated/src/ReanimatedModule/NativeReanimated.ts

```diff
@@ -13,11 +14,15 @@ import { getValueUnpackerCode } from '../valueUnpacker';
 import { isFabric } from '../PlatformChecker';
 import type React from 'react';
 import { getShadowNodeWrapperFromRef } from '../fabricUtils';
-import ReanimatedModule from '../specs/NativeReanimatedModule';
+import { ReanimatedTurboModule } from '../specs';
 import { ReanimatedError } from '../errors';
 
+export function createNativeReanimatedModule() {
+  return new NativeReanimatedModule();
+}
```

Comment:
What's the deal with wrapping constructor call with a factory function? Why can't we just instantiate `NativeReanimatedModule` directly instead of calling `createNativeReanimatedModule()`?

Reply:
I think it's the standard to use a dedicated factory function when dealing with the singleton pattern. I don't want to expose the `NativeReanimatedModule` class outside at all - you can see that `NativeReanimatedModule` class has no `export` keyword preceding it.